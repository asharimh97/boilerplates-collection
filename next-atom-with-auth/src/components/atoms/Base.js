import styled from "styled-components";
import {
  compose,
  layout,
  position,
  space,
  color,
  flexbox,
  grid,
  background,
  border,
  shadow,
  typography
} from "styled-system";
import propTypes from "@styled-system/prop-types";

import theme from "styles/theme";

const Base = styled("div")(
  compose(
    background,
    border,
    color,
    flexbox,
    grid,
    layout,
    position,
    shadow,
    space,
    typography
  )
);

Base.displayName = "Base";
Base.defaultProps = {
  theme
};

Base.propTypes = {
  ...propTypes.background,
  ...propTypes.border,
  ...propTypes.color,
  ...propTypes.flexbox,
  ...propTypes.grid,
  ...propTypes.layout,
  ...propTypes.position,
  ...propTypes.shadow,
  ...propTypes.space,
  ...propTypes.typography
};

export default Base;
