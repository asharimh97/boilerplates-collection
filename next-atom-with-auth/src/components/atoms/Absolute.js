import styled from "styled-components";
import Base from "./Base";

const Absolute = styled(Base)`
  position: absolute;
`;

export default Absolute;
