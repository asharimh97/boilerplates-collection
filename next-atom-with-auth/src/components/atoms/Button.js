import { Button as AntButton } from "antd";
import PropTypes from "prop-types";
import styled, { css } from "styled-components";
import { flexbox, layout, space, typography } from "styled-system";
import propTypes from "@styled-system/prop-types";

const flex = props => props.flex && { display: "flex" };

const color = props => {
  if (props.color) {
    if (props.type === "link" || props.type === "text") {
      return css`
        &,
        &:hover,
        &:active,
        &:focus {
          color: ${props.theme.colors[props.color]?.["primary"] || props.color};
        }
      `;
    }

    return css`
      background-color: ${props.theme.colors[props.color]?.["primary"] ||
      props.color};
      border-color: ${props.theme.colors[props.color]?.["primary"] ||
      props.color};
      color: white;

      &:hover,
      &:active,
      &:focus {
        background-color: ${props.theme.colors[props.color]?.["60"] ||
        props.color};
        border-color: ${props.theme.colors[props.color]?.["60"] || props.color};
        color: white;
      }
    `;
  }

  return null;
};

const Button = styled(AntButton)`
  ${color};

  ${flex};
  ${flexbox};
  ${layout};
  ${space};
  ${typography}
`;

Button.propTypes = {
  flex: PropTypes.bool,
  ...propTypes.flexbox,
  ...propTypes.layout,
  ...propTypes.space,
  ...propTypes.typography
};

Button.displayName = "Button";

export default Button;
