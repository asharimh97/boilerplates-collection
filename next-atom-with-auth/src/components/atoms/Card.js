import styled from "styled-components";
import Base from "./Base";

const Card = styled(Base)``;

Card.defaultProps = {
  background: "white",
  p: 3,
  borderRadius: "5px",
  boxShadow: 5
};

Card.displayName = "Card";

export default Card;
