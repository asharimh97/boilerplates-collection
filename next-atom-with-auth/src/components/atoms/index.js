export { default as Absolute } from "./Absolute";
export { default as Avatar } from "./Avatar";
export { default as Box } from "./Box";
export { default as Button } from "./Button";
export { default as Card } from "./Card";
export { default as Container } from "./Container";
export { default as Divider } from "./Divider";
export { default as Fixed } from "./Fixed";
export { default as Flex } from "./Flex";
export { default as Grid } from "./Grid";
export { default as Image } from "./Image";
export { default as Link } from "./Link";
export { default as Sticky } from "./Sticky";
export { default as Text } from "./Text";
export { default as Title } from "./Title";
