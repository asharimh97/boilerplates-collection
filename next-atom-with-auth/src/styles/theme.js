const createMediaQuery = n => `@media screen and (min-width:${n})`;

const addAliases = (arr, aliases) =>
  aliases.forEach((key, i) =>
    Object.defineProperty(arr, key, {
      enumerable: false,
      get() {
        return this[i];
      }
    })
  );

export const breakpoints = ["426px", "768px", "992px", "1200px"];

export const mediaQueries = breakpoints.map(createMediaQuery);

const aliases = ["sm", "md", "lg", "xl"];

addAliases(breakpoints, aliases);
addAliases(mediaQueries, aliases);
export const font = "Google Sans, Helvetica, Arial, sans-serif";

// this font size scaling follows styled-system standard
// however we can't change the standard scaling, theme will always return
// this sequence `[12, 14, 16, 20, 24, 32, 48]` even if we try to shift
// some values between the scales.
export const fontSizes = [12, 14, 16, 20, 24, 32, 48];
fontSizes.midSmall = 18;
fontSizes.mid = 28;
fontSizes.midBig = 40;

export const light = 300;
export const regular = 400;
export const medium = 500;
export const bold = 700;
export const fontWeights = {
  light,
  regular,
  medium,
  bold
};

// in this new spacing standard index `0` stands for 0 margin/padding
export const space = [0, 4, 8, 16, 24, 32, 40, 64, 128, 256, 512];
const shades = {
  black: {
    100: "#25282B",
    80: "#52575C",
    60: "#A0A4A8",
    40: "#CACCCF",
    20: "#EBEBEB",
    10: "#F5F5F5",
    5: "#FAFAFA",
    primary: "#25282B"
  },
  red: {
    80: "#99231D",
    60: "#cc2f26",
    50: "#ff3b30",
    40: "#ff8983",
    10: "#ffd8d6",
    primary: "#ff3b30"
  },
  green: {
    80: "#007a00",
    60: "#009b00",
    50: "#00b900",
    40: "#66d566",
    10: "#e5f8e5",
    primary: "#00b900"
  },
  orange: {
    80: "#c26307",
    60: "#e07b03",
    50: "#ff9500",
    40: "#ffc066",
    10: "#ffeacc",
    primary: "#ff9500"
  },
  blue: {
    80: "#02577e",
    60: "#039de3",
    50: "#1cb7fc",
    40: "#4fc7fd",
    10: "#e6f7ff",
    primary: "#1cb7fc"
  }
};

const textColor = {
  primary: shades.black[100],
  secondary: shades.black[80],
  tertiary: shades.black[60],
  quarternary: shades.black[40]
};

const inputColors = {
  success: shades.green,
  danger: shades.red,
  warning: shades.orange,
  disabled: shades.black,
  info: shades.blue
};

export const colors = {
  ...shades,
  ...inputColors,
  textColor,
  primary: shades.orange
};

export const radius = "3px";
export const shadows = [
  `0 1px 1px 0 rgba(10, 31, 68, 0.08), 0 0 1px 0 rgba(10, 31, 68, 0.08)`,
  `0 3px 4px 0 rgba(10, 31, 68, 0.1), 0 0 1px 0 rgba(10, 31, 68, 0.08)`,
  `0 8px 10px 0 rgba(10, 31, 68, 0.1), 0 0 1px 0 rgba(10, 31, 68, 0.08)`,
  `0 18px 18px 0 rgba(10, 31, 68, 0.12), 0 0 1px 0 rgba(10, 31, 68, 0.1)`,
  `0 26px 26px 0 rgba(10, 31, 68, 0.12), 0 0 1px 0 rgba(10, 31, 68, 0.1)`,
  `0 3px 32px 0 rgba(10, 31, 68, 0.1), 0 0 1px 0 rgba(10, 31, 68, 0.08)`
];

export const radii = ["3px", "5px"];
export const lineHeights = ["1", "1.5"];

export default {
  bold,
  breakpoints,
  colors,
  font,
  fontSizes,
  fontWeights,
  light,
  lineHeights,
  mediaQueries,
  medium,
  radii,
  radius,
  regular,
  shadows,
  space
};
