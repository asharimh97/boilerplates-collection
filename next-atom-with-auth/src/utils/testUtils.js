import { createStore, applyMiddleware } from "redux";

import { middlewares } from "stores";
import reducers from "stores/reducers";
import { shallow } from "enzyme";

import { render as rtlRender } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { create as rtrCreate } from "react-test-renderer";
import "@testing-library/jest-dom/extend-expect";
import "jest-styled-components";

export const render = (component, options) => rtlRender(component, options);

export const create = component => rtrCreate(component);

export const fireEvent = userEvent;

/**
 * Function to find a component with `data-test` attribute
 * @param {ShallowWrapper} wrapper - Shallowed component.
 * @param {String} val -  `data-test` attribute value.
 * @return {ShallowWrapper}
 */
export const findByTestAttribute = (wrapper, val) => {
  return wrapper.find(`[data-test="${val}"]`);
};

/**
 * Function to validate/test if component is appear on component
 * @param {ShallowWrapper} wrapper - Shallowed component.
 * @param {String} componentName - component name to look out.
 */
export const findComponent = (wrapper, componentName) => {
  const comp = findByTestAttribute(wrapper, componentName);
  expect(comp.length).toBe(1);
};

/**
 * Function to validate/test if component is appear on component
 * @param {ShallowWrapper} wrapper - Shallowed component.
 * @param {String} componentName - component name to look out based on data-test.
 * @param {Number} total - expected total
 */
export const findComponentTotal = (wrapper, componentName, total) => {
  const comp = findByTestAttribute(wrapper, componentName);
  expect(comp.length).toBe(total);
};

/**
 * Function to validate if component is not found/appear in tested component
 * @param {ShallowWrapper} wrapper - Shallowed component.
 * @param {String} componentName - component name to look out.
 */
export const unFindComponent = (wrapper, componentName) => {
  const comp = findByTestAttribute(wrapper, componentName);
  expect(comp.length).toBe(0);
};

/**
 * Function to validate if action is exist in connected component
 * @param {ShallowWrapper} wrapper - Shallowed component
 * @param {String} actionName - action name
 */
export const findActionProp = (wrapper, actionName) => {
  const actionProp = wrapper.instance().props[actionName];
  expect(actionProp).toBeInstanceOf(Function);
};

/**
 * Create a testing store with imported reducers, middleware, and state
 *  globals: reducers
 * @param {Object} state - State for the store
 * @function storeFactory
 * @return {Store}
 */
export const storeFactory = state => {
  const storeWithMiddleware = createStore(
    reducers,
    state,
    applyMiddleware(...middlewares)
  );
  return storeWithMiddleware;
};

/**
 * Mock next route since next route is meant to be in client side
 * while testing component which uses next route is tested in
 * server side.
 */
export const mockedNextRouter = {
  pushRoute: () => {},
  route: "",
  push: () => {},
  replace: () => {}
};

/**
 * Function to shorten shallowing unconnected component
 * @param {Node} Component - Component being shallowed
 * @param {Object} props - properties
 * @param {Object} state - state given to component
 * @return {ShallowWrapper}
 */
export const shallowComponent = (Component, props = {}, state = null) => {
  const wrapper = shallow(<Component {...props} />);

  if (state) wrapper.setState(state);

  return wrapper;
};

/**
 * Function to shorten shallowing redux-connected component non-page and non-form
 * @param {*} Component - Component being shallowed
 * @param {*} state - redux state value
 * @param {*} props - additional prop
 * @return {ShallowWrapper}
 */
export const shallowReduxComponent = (Component, state = {}, props = {}) => {
  const store = storeFactory(state);
  const wrapper = shallow(<Component store={store} {...props} />)
    .dive()
    .dive();

  return wrapper;
};

/**
 * Function to shorten shallowing redux-connected page/form component
 * @param {*} Component - Component being shallowed
 * @param {*} state - redux state value
 * @param {*} props - additional prop
 * @return {ShallowWrapper}
 */
export const shallowConnect = (Component, state = {}, props = {}) => {
  const store = storeFactory(state);
  const wrapper = shallow(<Component store={store} {...props} />)
    .dive()
    .dive()
    .dive();

  return wrapper;
};
