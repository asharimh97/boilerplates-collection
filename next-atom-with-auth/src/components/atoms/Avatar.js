import styled, { css } from "styled-components";
import { convertNumberToPixel } from "utils/helper";
import Image from "./Image";

const sizes = {
  small: { width: "40px", height: "40px" },
  regular: { width: "50px", height: "50px" },
  big: { width: "65px", height: "65px" }
};

const Avatar = styled(Image).attrs(props => ({
  autofit: true,
  height: sizes[props.size]?.width || convertNumberToPixel(props.size),
  fullRounded: props.circle
}))`
  ${props =>
    props.size && !Number.isInteger(props.size)
      ? css`
          ${sizes[props.size]}
        `
      : css`
          width: ${props.size};
          height: ${props.size};
        `}
`;

Avatar.propTypes = {};

Avatar.displayName = "Avatar";

Avatar.defaultProps = {
  size: "regular",
  src: "/static/images/avatar.jpeg"
};

export default Avatar;
