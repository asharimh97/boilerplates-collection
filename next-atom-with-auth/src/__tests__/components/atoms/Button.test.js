import { Button } from "components/atoms";
import theme from "styles/theme";
import { create, fireEvent, render } from "utils/testUtils";

test("renders `Button` withtout error", () => {
  const handleClick = jest.fn();
  const { getByRole } = render(<Button onClick={handleClick} />);

  fireEvent.click(getByRole("button"));

  expect(getByRole("button")).toBeInTheDocument();
  expect(handleClick).toBeCalled();
});

const setup = props => create(<Button {...props} />).toJSON();

describe("test render styled `Button`", () => {
  const props = {
    theme
  };
  test("renders `Button` with color", () => {
    const button = setup({ ...props, color: "red" });
    expect(button).toMatchSnapshot();
    expect(button).toHaveStyleRule(
      "background-color",
      theme.colors.red.primary
    );
    expect(button).toHaveStyleRule("border-color", theme.colors.red.primary);
    expect(button).toHaveStyleRule("color", "white");
  });
  test("renders `Button` with hex color", () => {
    const button = setup({ ...props, color: "#f0f0f0" });
    expect(button).toMatchSnapshot();
    expect(button).toHaveStyleRule("background-color", "#f0f0f0");
    expect(button).toHaveStyleRule("border-color", "#f0f0f0");
    expect(button).toHaveStyleRule("color", "white");
  });

  test("renders `Button` with `type='link'`", () => {
    const button = setup({ ...props, type: "link", color: "red" });
    expect(button).toMatchSnapshot();
    expect(button).toHaveStyleRule("color", theme.colors.red.primary);
  });
  test("renders `Button` with `type='link'` and hex color", () => {
    const button = setup({ ...props, type: "link", color: "#f0f0f0" });
    expect(button).toMatchSnapshot();
    expect(button).toHaveStyleRule("color", "#f0f0f0");
  });
});
