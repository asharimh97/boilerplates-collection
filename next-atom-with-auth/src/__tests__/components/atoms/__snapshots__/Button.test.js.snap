// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test render styled \`Button\` renders \`Button\` with \`type='link'\` 1`] = `
.c0,
.c0:hover,
.c0:active,
.c0:focus {
  color: #ff3b30;
}

<button
  className="ant-btn ant-btn-link c0"
  color="red"
  onClick={[Function]}
  theme={
    Object {
      "bold": 700,
      "breakpoints": Array [
        "426px",
        "768px",
        "992px",
        "1200px",
      ],
      "colors": Object {
        "black": Object {
          "10": "#F5F5F5",
          "100": "#25282B",
          "20": "#EBEBEB",
          "40": "#CACCCF",
          "5": "#FAFAFA",
          "60": "#A0A4A8",
          "80": "#52575C",
          "primary": "#25282B",
        },
        "blue": Object {
          "10": "#e6f7ff",
          "40": "#4fc7fd",
          "50": "#1cb7fc",
          "60": "#039de3",
          "80": "#02577e",
          "primary": "#1cb7fc",
        },
        "danger": Object {
          "10": "#ffd8d6",
          "40": "#ff8983",
          "50": "#ff3b30",
          "60": "#cc2f26",
          "80": "#99231D",
          "primary": "#ff3b30",
        },
        "disabled": Object {
          "10": "#F5F5F5",
          "100": "#25282B",
          "20": "#EBEBEB",
          "40": "#CACCCF",
          "5": "#FAFAFA",
          "60": "#A0A4A8",
          "80": "#52575C",
          "primary": "#25282B",
        },
        "green": Object {
          "10": "#e5f8e5",
          "40": "#66d566",
          "50": "#00b900",
          "60": "#009b00",
          "80": "#007a00",
          "primary": "#00b900",
        },
        "info": Object {
          "10": "#e6f7ff",
          "40": "#4fc7fd",
          "50": "#1cb7fc",
          "60": "#039de3",
          "80": "#02577e",
          "primary": "#1cb7fc",
        },
        "orange": Object {
          "10": "#ffeacc",
          "40": "#ffc066",
          "50": "#ff9500",
          "60": "#e07b03",
          "80": "#c26307",
          "primary": "#ff9500",
        },
        "primary": Object {
          "10": "#ffeacc",
          "40": "#ffc066",
          "50": "#ff9500",
          "60": "#e07b03",
          "80": "#c26307",
          "primary": "#ff9500",
        },
        "red": Object {
          "10": "#ffd8d6",
          "40": "#ff8983",
          "50": "#ff3b30",
          "60": "#cc2f26",
          "80": "#99231D",
          "primary": "#ff3b30",
        },
        "success": Object {
          "10": "#e5f8e5",
          "40": "#66d566",
          "50": "#00b900",
          "60": "#009b00",
          "80": "#007a00",
          "primary": "#00b900",
        },
        "textColor": Object {
          "primary": "#25282B",
          "quarternary": "#CACCCF",
          "secondary": "#52575C",
          "tertiary": "#A0A4A8",
        },
        "warning": Object {
          "10": "#ffeacc",
          "40": "#ffc066",
          "50": "#ff9500",
          "60": "#e07b03",
          "80": "#c26307",
          "primary": "#ff9500",
        },
      },
      "font": "Google Sans, Helvetica, Arial, sans-serif",
      "fontSizes": Array [
        "0.8em",
        "1em",
        "1.25em",
        "1.563em",
        "1.953em",
        "2.441em",
        "3.052em",
      ],
      "fontWeights": Object {
        "bold": 700,
        "light": 300,
        "medium": 500,
        "regular": 400,
        "semibold": 600,
      },
      "light": 300,
      "mediaQueries": Array [
        "@media screen and (min-width:426px)",
        "@media screen and (min-width:768px)",
        "@media screen and (min-width:992px)",
        "@media screen and (min-width:1200px)",
      ],
      "medium": 500,
      "regular": 400,
      "semibold": 600,
      "shadows": Array [
        "0 1px 1px 0 rgba(10, 31, 68, 0.08), 0 0 1px 0 rgba(10, 31, 68, 0.08)",
        "0 3px 4px 0 rgba(10, 31, 68, 0.1), 0 0 1px 0 rgba(10, 31, 68, 0.08)",
        "0 8px 10px 0 rgba(10, 31, 68, 0.1), 0 0 1px 0 rgba(10, 31, 68, 0.08)",
        "0 18px 18px 0 rgba(10, 31, 68, 0.12), 0 0 1px 0 rgba(10, 31, 68, 0.1)",
        "0 26px 26px 0 rgba(10, 31, 68, 0.12), 0 0 1px 0 rgba(10, 31, 68, 0.1)",
        "0 3px 32px 0 rgba(10, 31, 68, 0.1), 0 0 1px 0 rgba(10, 31, 68, 0.08)",
      ],
      "space": Array [
        0,
        4,
        8,
        16,
        24,
        32,
        40,
        64,
        128,
        256,
        512,
      ],
    }
  }
  type="button"
/>
`;

exports[`test render styled \`Button\` renders \`Button\` with \`type='link'\` and hex color 1`] = `
.c0,
.c0:hover,
.c0:active,
.c0:focus {
  color: #f0f0f0;
}

<button
  className="ant-btn ant-btn-link c0"
  color="#f0f0f0"
  onClick={[Function]}
  theme={
    Object {
      "bold": 700,
      "breakpoints": Array [
        "426px",
        "768px",
        "992px",
        "1200px",
      ],
      "colors": Object {
        "black": Object {
          "10": "#F5F5F5",
          "100": "#25282B",
          "20": "#EBEBEB",
          "40": "#CACCCF",
          "5": "#FAFAFA",
          "60": "#A0A4A8",
          "80": "#52575C",
          "primary": "#25282B",
        },
        "blue": Object {
          "10": "#e6f7ff",
          "40": "#4fc7fd",
          "50": "#1cb7fc",
          "60": "#039de3",
          "80": "#02577e",
          "primary": "#1cb7fc",
        },
        "danger": Object {
          "10": "#ffd8d6",
          "40": "#ff8983",
          "50": "#ff3b30",
          "60": "#cc2f26",
          "80": "#99231D",
          "primary": "#ff3b30",
        },
        "disabled": Object {
          "10": "#F5F5F5",
          "100": "#25282B",
          "20": "#EBEBEB",
          "40": "#CACCCF",
          "5": "#FAFAFA",
          "60": "#A0A4A8",
          "80": "#52575C",
          "primary": "#25282B",
        },
        "green": Object {
          "10": "#e5f8e5",
          "40": "#66d566",
          "50": "#00b900",
          "60": "#009b00",
          "80": "#007a00",
          "primary": "#00b900",
        },
        "info": Object {
          "10": "#e6f7ff",
          "40": "#4fc7fd",
          "50": "#1cb7fc",
          "60": "#039de3",
          "80": "#02577e",
          "primary": "#1cb7fc",
        },
        "orange": Object {
          "10": "#ffeacc",
          "40": "#ffc066",
          "50": "#ff9500",
          "60": "#e07b03",
          "80": "#c26307",
          "primary": "#ff9500",
        },
        "primary": Object {
          "10": "#ffeacc",
          "40": "#ffc066",
          "50": "#ff9500",
          "60": "#e07b03",
          "80": "#c26307",
          "primary": "#ff9500",
        },
        "red": Object {
          "10": "#ffd8d6",
          "40": "#ff8983",
          "50": "#ff3b30",
          "60": "#cc2f26",
          "80": "#99231D",
          "primary": "#ff3b30",
        },
        "success": Object {
          "10": "#e5f8e5",
          "40": "#66d566",
          "50": "#00b900",
          "60": "#009b00",
          "80": "#007a00",
          "primary": "#00b900",
        },
        "textColor": Object {
          "primary": "#25282B",
          "quarternary": "#CACCCF",
          "secondary": "#52575C",
          "tertiary": "#A0A4A8",
        },
        "warning": Object {
          "10": "#ffeacc",
          "40": "#ffc066",
          "50": "#ff9500",
          "60": "#e07b03",
          "80": "#c26307",
          "primary": "#ff9500",
        },
      },
      "font": "Google Sans, Helvetica, Arial, sans-serif",
      "fontSizes": Array [
        "0.8em",
        "1em",
        "1.25em",
        "1.563em",
        "1.953em",
        "2.441em",
        "3.052em",
      ],
      "fontWeights": Object {
        "bold": 700,
        "light": 300,
        "medium": 500,
        "regular": 400,
        "semibold": 600,
      },
      "light": 300,
      "mediaQueries": Array [
        "@media screen and (min-width:426px)",
        "@media screen and (min-width:768px)",
        "@media screen and (min-width:992px)",
        "@media screen and (min-width:1200px)",
      ],
      "medium": 500,
      "regular": 400,
      "semibold": 600,
      "shadows": Array [
        "0 1px 1px 0 rgba(10, 31, 68, 0.08), 0 0 1px 0 rgba(10, 31, 68, 0.08)",
        "0 3px 4px 0 rgba(10, 31, 68, 0.1), 0 0 1px 0 rgba(10, 31, 68, 0.08)",
        "0 8px 10px 0 rgba(10, 31, 68, 0.1), 0 0 1px 0 rgba(10, 31, 68, 0.08)",
        "0 18px 18px 0 rgba(10, 31, 68, 0.12), 0 0 1px 0 rgba(10, 31, 68, 0.1)",
        "0 26px 26px 0 rgba(10, 31, 68, 0.12), 0 0 1px 0 rgba(10, 31, 68, 0.1)",
        "0 3px 32px 0 rgba(10, 31, 68, 0.1), 0 0 1px 0 rgba(10, 31, 68, 0.08)",
      ],
      "space": Array [
        0,
        4,
        8,
        16,
        24,
        32,
        40,
        64,
        128,
        256,
        512,
      ],
    }
  }
  type="button"
/>
`;

exports[`test render styled \`Button\` renders \`Button\` with color 1`] = `
.c0 {
  background-color: #ff3b30;
  border-color: #ff3b30;
  color: white;
}

.c0:hover,
.c0:active,
.c0:focus {
  background-color: #cc2f26;
  border-color: #cc2f26;
  color: white;
}

<button
  className="ant-btn c0"
  color="red"
  onClick={[Function]}
  theme={
    Object {
      "bold": 700,
      "breakpoints": Array [
        "426px",
        "768px",
        "992px",
        "1200px",
      ],
      "colors": Object {
        "black": Object {
          "10": "#F5F5F5",
          "100": "#25282B",
          "20": "#EBEBEB",
          "40": "#CACCCF",
          "5": "#FAFAFA",
          "60": "#A0A4A8",
          "80": "#52575C",
          "primary": "#25282B",
        },
        "blue": Object {
          "10": "#e6f7ff",
          "40": "#4fc7fd",
          "50": "#1cb7fc",
          "60": "#039de3",
          "80": "#02577e",
          "primary": "#1cb7fc",
        },
        "danger": Object {
          "10": "#ffd8d6",
          "40": "#ff8983",
          "50": "#ff3b30",
          "60": "#cc2f26",
          "80": "#99231D",
          "primary": "#ff3b30",
        },
        "disabled": Object {
          "10": "#F5F5F5",
          "100": "#25282B",
          "20": "#EBEBEB",
          "40": "#CACCCF",
          "5": "#FAFAFA",
          "60": "#A0A4A8",
          "80": "#52575C",
          "primary": "#25282B",
        },
        "green": Object {
          "10": "#e5f8e5",
          "40": "#66d566",
          "50": "#00b900",
          "60": "#009b00",
          "80": "#007a00",
          "primary": "#00b900",
        },
        "info": Object {
          "10": "#e6f7ff",
          "40": "#4fc7fd",
          "50": "#1cb7fc",
          "60": "#039de3",
          "80": "#02577e",
          "primary": "#1cb7fc",
        },
        "orange": Object {
          "10": "#ffeacc",
          "40": "#ffc066",
          "50": "#ff9500",
          "60": "#e07b03",
          "80": "#c26307",
          "primary": "#ff9500",
        },
        "primary": Object {
          "10": "#ffeacc",
          "40": "#ffc066",
          "50": "#ff9500",
          "60": "#e07b03",
          "80": "#c26307",
          "primary": "#ff9500",
        },
        "red": Object {
          "10": "#ffd8d6",
          "40": "#ff8983",
          "50": "#ff3b30",
          "60": "#cc2f26",
          "80": "#99231D",
          "primary": "#ff3b30",
        },
        "success": Object {
          "10": "#e5f8e5",
          "40": "#66d566",
          "50": "#00b900",
          "60": "#009b00",
          "80": "#007a00",
          "primary": "#00b900",
        },
        "textColor": Object {
          "primary": "#25282B",
          "quarternary": "#CACCCF",
          "secondary": "#52575C",
          "tertiary": "#A0A4A8",
        },
        "warning": Object {
          "10": "#ffeacc",
          "40": "#ffc066",
          "50": "#ff9500",
          "60": "#e07b03",
          "80": "#c26307",
          "primary": "#ff9500",
        },
      },
      "font": "Google Sans, Helvetica, Arial, sans-serif",
      "fontSizes": Array [
        "0.8em",
        "1em",
        "1.25em",
        "1.563em",
        "1.953em",
        "2.441em",
        "3.052em",
      ],
      "fontWeights": Object {
        "bold": 700,
        "light": 300,
        "medium": 500,
        "regular": 400,
        "semibold": 600,
      },
      "light": 300,
      "mediaQueries": Array [
        "@media screen and (min-width:426px)",
        "@media screen and (min-width:768px)",
        "@media screen and (min-width:992px)",
        "@media screen and (min-width:1200px)",
      ],
      "medium": 500,
      "regular": 400,
      "semibold": 600,
      "shadows": Array [
        "0 1px 1px 0 rgba(10, 31, 68, 0.08), 0 0 1px 0 rgba(10, 31, 68, 0.08)",
        "0 3px 4px 0 rgba(10, 31, 68, 0.1), 0 0 1px 0 rgba(10, 31, 68, 0.08)",
        "0 8px 10px 0 rgba(10, 31, 68, 0.1), 0 0 1px 0 rgba(10, 31, 68, 0.08)",
        "0 18px 18px 0 rgba(10, 31, 68, 0.12), 0 0 1px 0 rgba(10, 31, 68, 0.1)",
        "0 26px 26px 0 rgba(10, 31, 68, 0.12), 0 0 1px 0 rgba(10, 31, 68, 0.1)",
        "0 3px 32px 0 rgba(10, 31, 68, 0.1), 0 0 1px 0 rgba(10, 31, 68, 0.08)",
      ],
      "space": Array [
        0,
        4,
        8,
        16,
        24,
        32,
        40,
        64,
        128,
        256,
        512,
      ],
    }
  }
  type="button"
/>
`;

exports[`test render styled \`Button\` renders \`Button\` with hex color 1`] = `
.c0 {
  background-color: #f0f0f0;
  border-color: #f0f0f0;
  color: white;
}

.c0:hover,
.c0:active,
.c0:focus {
  background-color: #f0f0f0;
  border-color: #f0f0f0;
  color: white;
}

<button
  className="ant-btn c0"
  color="#f0f0f0"
  onClick={[Function]}
  theme={
    Object {
      "bold": 700,
      "breakpoints": Array [
        "426px",
        "768px",
        "992px",
        "1200px",
      ],
      "colors": Object {
        "black": Object {
          "10": "#F5F5F5",
          "100": "#25282B",
          "20": "#EBEBEB",
          "40": "#CACCCF",
          "5": "#FAFAFA",
          "60": "#A0A4A8",
          "80": "#52575C",
          "primary": "#25282B",
        },
        "blue": Object {
          "10": "#e6f7ff",
          "40": "#4fc7fd",
          "50": "#1cb7fc",
          "60": "#039de3",
          "80": "#02577e",
          "primary": "#1cb7fc",
        },
        "danger": Object {
          "10": "#ffd8d6",
          "40": "#ff8983",
          "50": "#ff3b30",
          "60": "#cc2f26",
          "80": "#99231D",
          "primary": "#ff3b30",
        },
        "disabled": Object {
          "10": "#F5F5F5",
          "100": "#25282B",
          "20": "#EBEBEB",
          "40": "#CACCCF",
          "5": "#FAFAFA",
          "60": "#A0A4A8",
          "80": "#52575C",
          "primary": "#25282B",
        },
        "green": Object {
          "10": "#e5f8e5",
          "40": "#66d566",
          "50": "#00b900",
          "60": "#009b00",
          "80": "#007a00",
          "primary": "#00b900",
        },
        "info": Object {
          "10": "#e6f7ff",
          "40": "#4fc7fd",
          "50": "#1cb7fc",
          "60": "#039de3",
          "80": "#02577e",
          "primary": "#1cb7fc",
        },
        "orange": Object {
          "10": "#ffeacc",
          "40": "#ffc066",
          "50": "#ff9500",
          "60": "#e07b03",
          "80": "#c26307",
          "primary": "#ff9500",
        },
        "primary": Object {
          "10": "#ffeacc",
          "40": "#ffc066",
          "50": "#ff9500",
          "60": "#e07b03",
          "80": "#c26307",
          "primary": "#ff9500",
        },
        "red": Object {
          "10": "#ffd8d6",
          "40": "#ff8983",
          "50": "#ff3b30",
          "60": "#cc2f26",
          "80": "#99231D",
          "primary": "#ff3b30",
        },
        "success": Object {
          "10": "#e5f8e5",
          "40": "#66d566",
          "50": "#00b900",
          "60": "#009b00",
          "80": "#007a00",
          "primary": "#00b900",
        },
        "textColor": Object {
          "primary": "#25282B",
          "quarternary": "#CACCCF",
          "secondary": "#52575C",
          "tertiary": "#A0A4A8",
        },
        "warning": Object {
          "10": "#ffeacc",
          "40": "#ffc066",
          "50": "#ff9500",
          "60": "#e07b03",
          "80": "#c26307",
          "primary": "#ff9500",
        },
      },
      "font": "Google Sans, Helvetica, Arial, sans-serif",
      "fontSizes": Array [
        "0.8em",
        "1em",
        "1.25em",
        "1.563em",
        "1.953em",
        "2.441em",
        "3.052em",
      ],
      "fontWeights": Object {
        "bold": 700,
        "light": 300,
        "medium": 500,
        "regular": 400,
        "semibold": 600,
      },
      "light": 300,
      "mediaQueries": Array [
        "@media screen and (min-width:426px)",
        "@media screen and (min-width:768px)",
        "@media screen and (min-width:992px)",
        "@media screen and (min-width:1200px)",
      ],
      "medium": 500,
      "regular": 400,
      "semibold": 600,
      "shadows": Array [
        "0 1px 1px 0 rgba(10, 31, 68, 0.08), 0 0 1px 0 rgba(10, 31, 68, 0.08)",
        "0 3px 4px 0 rgba(10, 31, 68, 0.1), 0 0 1px 0 rgba(10, 31, 68, 0.08)",
        "0 8px 10px 0 rgba(10, 31, 68, 0.1), 0 0 1px 0 rgba(10, 31, 68, 0.08)",
        "0 18px 18px 0 rgba(10, 31, 68, 0.12), 0 0 1px 0 rgba(10, 31, 68, 0.1)",
        "0 26px 26px 0 rgba(10, 31, 68, 0.12), 0 0 1px 0 rgba(10, 31, 68, 0.1)",
        "0 3px 32px 0 rgba(10, 31, 68, 0.1), 0 0 1px 0 rgba(10, 31, 68, 0.08)",
      ],
      "space": Array [
        0,
        4,
        8,
        16,
        24,
        32,
        40,
        64,
        128,
        256,
        512,
      ],
    }
  }
  type="button"
/>
`;
